backend/tests/functional/branch/test_graphql_mutation.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pytest
+from infrahub_sdk.exceptions import BranchNotFoundError
+from infrahub_sdk.graphql import Mutation
+
+from infrahub.core.constants import InfrahubKind
+from infrahub.core.node import Node
+from infrahub.services.adapters.cache.redis import RedisCache
+from tests.constants import TestKind
+from tests.helpers.file_repo import FileRepo
+from tests.helpers.schema import CAR_SCHEMA, load_schema
+from tests.helpers.test_app import TestInfrahubApp
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from infrahub_sdk import InfrahubClient
+
+    from infrahub.database import InfrahubDatabase
+    from tests.adapters.message_bus import BusSimulator
+
+
+class TestBranchMutations(TestInfrahubApp):
+    @pytest.fixture(scope="class")
+    async def initial_dataset(
+        self,
+        db: InfrahubDatabase,
+        initialize_registry: None,
+        git_repos_source_dir_module_scope: Path,
+        client: InfrahubClient,
+        bus_simulator: BusSimulator,
+        prefect_test_fixture: None,
+    ) -> str:
+        await load_schema(db, schema=CAR_SCHEMA)
+
+        john = await Node.init(schema=TestKind.PERSON, db=db)
+        await john.new(db=db, name="John", height=175, description="The famous Joe Doe")
+        await john.save(db=db)
+
+        koenigsegg = await Node.init(schema=TestKind.MANUFACTURER, db=db)
+        await koenigsegg.new(db=db, name="Koenigsegg")
+        await koenigsegg.save(db=db)
+
+        people = await Node.init(schema=InfrahubKind.STANDARDGROUP, db=db)
+        await people.new(db=db, name="people", members=[john])
+        await people.save(db=db)
+
+        jesko = await Node.init(schema=TestKind.CAR, db=db)
+        await jesko.new(
+            db=db,
+            name="Jesko",
+            color="Red",
+            description="A limited production mid-engine sports car",
+            owner=john,
+            manufacturer=koenigsegg,
+        )
+        await jesko.save(db=db)
+
+        bus_simulator.service.cache = RedisCache()
```

Comment:
I don't have another solution in mind just now but I think we should have some other solution for this in the future. Could be something that we look at when reworking the tests in general to make them easier to use. Might be here that we should just have different fixtures for different type of buses. Not a blocker now more of an observation.